example/ghost/t8_test_ghost_large_level_diff.cxx

```diff
@@ -0,0 +1,235 @@
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <sc_flops.h>
+#include <sc_statistics.h>
+#include <sc_options.h>
+#include <t8_cmesh_readmshfile.h>
+#include <t8_cmesh_vtk.h>
+#include <t8_default/t8_dprism.h>
+#include <t8_default/t8_dtet.h>
+#include <t8_forest/t8_forest_adapt.h>
+#include <t8_default_cxx.hxx>
+
+
+/*The refinement creterion
+ * returns 1 if we refine the element, 0 otherwise.
+ */
+static int
+t8_ghost_fractal_adapt (t8_forest_t forest, t8_forest_t forest_from,
+                            t8_locidx_t which_tree, t8_locidx_t lelement_id,
+                            t8_eclass_scheme_c * ts, int num_elements,
+                            t8_element_t * elements[])
+{
+  int                   level;
+  int                   type, child_id;
+  T8_ASSERT (num_elements == 1 || num_elements ==
+             ts->t8_element_num_children (elements[0]));
+
+  level = ts->t8_element_level (elements[0]);
+  if (level >= *(int *) t8_forest_get_user_data (forest)) {
+    return 0;
+  }
+  if(ts->eclass == T8_ECLASS_PRISM){
+      type = ((t8_dprism_t *) elements[0])->tri.type;
+      /* refine type 0 except those with child_id 3 or 4*/
+      if (type == 0) {
+        child_id = ts->t8_element_child_id(elements[0]);
+        /*Not refining*/
+        if(child_id == 3 || child_id == 4){
+            return 0;
+        }
+        /*Refining*/
+        else{
+            return 1;
+        }
+      }
+      return 0;
+  }
+  else if(ts->eclass == T8_ECLASS_TET){
+      type = ((t8_dtet_t * ) elements[0])->type;
+      /*Refine tets of type 0, 3 or 5*/
+      if (type == 0 || type == 3 || type == 5){
+          return 1;
+      }
+      return 0;
+  }
+  else if(ts->eclass == T8_ECLASS_HEX){
+      child_id = ts->t8_element_child_id(elements[0]);
+      if(child_id == 0 || child_id == 3 || child_id == 5 || child_id == 6){
+          return 1;
+      };
+      return 0;
+  }
+
+}
+
+/**
+ * Creates a forest given a meshfile. If no meshfile is given, a hypercube of
+ * two prisms is created. The forest is refined adaptively, partitioned and the
+ * ghost elements are computed.
+ *
+ * \param[in] prefix  The prefix to the meshfile
+ * @param dim       Dimension of the mesh
+ * @param level     Initial refinement level
+ * @param refine    Number of levels the forest will be refined
+ * @param no_vtk    Flag for vtk-output
+ * @param comm      The MPI-communicator
+ */
+static void
+t8_ghost_large_level_diff(const char* prefix, int dim, int level, int refine,
+                          int no_vtk, sc_MPI_Comm comm){
+    t8_forest_t     forest, forest_adapt, forest_partition;
+    t8_cmesh_t      cmesh, cmesh_partition;
+    sc_flopinfo_t       fi, snapshot;
+    sc_statinfo_t       stats[1];
+
+
+    if(prefix != NULL){
+        cmesh = t8_cmesh_from_msh_file((char *) prefix, 1, comm, dim, 0);
+    }
+    /*If no prefix given, create hypercube*/
```

Comment:
Comments look much nicer with spaces after/before *.
/* like this */ 
/*not like this*/

Reply:
I changed this.